src/libraries/System.Threading.Thread/tests/ThreadTests.cs

```diff
@@ -647,6 +647,19 @@ public static void NameTest()
             });
         }
 
+        [Fact]
+        public static void ThreadNameDoesNotAffectProcessName()
+        {
+            // On Linux, changing the main thread name affects ProcessName.
+            // To avoid that, .NET ignores requests to change the main thread name.
+            RemoteExecutor.Invoke(() =>
+            {
+                const string threadName = "my-thread";
+                Thread.CurrentThread.Name = threadName;
```

Comment:
Nit, it's a bit pedantic, but this doesn't verify that the thread name didn't change to something else (eg empty). You might save it before the set , and verify it after. Instead of asserting not equal.

Reply:
`Thread.Name` is a write-once property that gets stored in a field. It never reads something from the OS.

Reply:
I was essentially making the same suggestion as @stephentoub

Reply:
I realized when I was giving similar replies :smile: